geode-core/src/main/java/org/apache/geode/cache/client/internal/QueueManagerImpl.java

```diff
@@ -805,10 +815,13 @@ private QueueConnectionImpl createNewPrimary(Set<ServerLocation> excludedServers
       excludedServers.addAll(servers);
     }
 
+    return primary;
+  }
+
+  public void markQueueAsReadyForEvents(QueueConnectionImpl primary) {
```

Comment:
Perhaps using `@NotNull` annotations on the API. Make the caller guarantee the `primary` parameter is not `null`.

Reply:
done